orttraining/orttraining/training_api/utils.h

```diff
@@ -10,6 +10,31 @@
 namespace onnxruntime {
 namespace training {
 namespace api {
+
+struct ModelIdentifiers {
+  // ModelIdentifiers struct enables an easy way to store and identify the models used for training, evaluation
+  // and optimization.
+  // The model can be specified by a path to the model file or by a span of bytes containing the model data. However,
+  // only one of these two options can be used at a time.
+  // Training Session creation api enforces this by allowing the user to specify only one of the two options.
+  const std::optional<std::string> train_model_path, eval_model, optim_model = std::nullopt;
+  gsl::span<const uint8_t> train_model_data;
+  gsl::span<const uint8_t> eval_model_data;
+  gsl::span<const uint8_t> optim_model_data;
```

Comment:
Optional: Maybe you could use some form of a union (std::variant for example) instead of having two variables per model?